src/protocols/user_directed_commissioning/UserDirectedCommissioning.h

```diff
@@ -0,0 +1,192 @@
+                              chip::Inet::IPAddress commissioner, uint16_t port);
+};
+
+class DLL_EXPORT UserDirectedCommissioningServer : public TransportMgrDelegate
+{
+public:
+    static UserDirectedCommissioningServer & GetInstance()
+    {
+        static UserDirectedCommissioningServer instance;
+        return instance;
+    }
+
+    /**
+     * Set the listener to be called when a UDC request is received
+     * and the Instance Name provided needs to be resolved.
+     *
+     * The resolver should call OnCommissionableNodeFound when the instance is found
+     *
+     *  @param[in]    instanceNameResolver    The callback function to receive UDC request instance name.
+     *
+     */
+    void SetInstanceNameResolver(InstanceNameResolver * instanceNameResolver) { mInstanceNameResolver = instanceNameResolver; }
+
+    /**
+     * Set the listener to be called when a UDC request is received
+     * and the Instance Name has been resolved.
+     *
+     * The provider should prompt the user to allow commissioning of the node and provide the setup code.
+     *
+     *  @param[in]    userConfirmationProvider    The callback function to obtain user confirmation.
+     *
+     */
+    void SetUserConfirmationProvider(UserConfirmationProvider * userConfirmationProvider)
+    {
+        mUserConfirmationProvider = userConfirmationProvider;
+    }
+
+    /**
+     * Update the processing state for a UDC Client based upon instance name.
+     *
+     * This can be used by the UX to set the state to one of the following values:
+     * - kUserDeclined
+     * - kObtainingOnboardingPayload
+     * - kCommissioningNode
+     * - kCommissioningFailed
+     *
+     *  @param[in]    instanceName    The instance name for the UDC Client.
+     *  @param[in]    state           The state for the UDC Client.
+     *
+     */
+    void SetUDCClientProcessingState(char * instanceName, UDCClientProcessingState state);
+
+    /**
+     * Reset the processing states for all UDC Clients
+     *
+     */
+    void ResetUDCClientProcessingStates() { mUdcClients.ResetUDCClientStates(); }
+
+    /**
+     * Called when a CHIP Node in commissioning mode is found.
+     *
+     * Lookup instanceName from nodeData in the active UDC Client states
+     * and if current state is kDiscoveringNode then change to kPromptingUser and
+     * call UX Prompt callback
+     *
+     *  @param[in]    nodeData        DNS-SD response data.
+     *
+     */
+    void OnCommissionableNodeFound(const Mdns::DiscoveredNodeData & nodeData);
+
+private:
+    InstanceNameResolver * mInstanceNameResolver         = nullptr;
+    UserConfirmationProvider * mUserConfirmationProvider = nullptr;
+
+    void OnMessageReceived(const Transport::PeerAddress & source, System::PacketBufferHandle && msgBuf) override;
+
+    UDCClients<CHIP_CONFIG_PEER_CONNECTION_POOL_SIZE> mUdcClients; // < Active UDC clients
```

Comment:
Why is that the right pool size?

Reply:
Fixed this copy/paste error and added a comment about the source of the size